test/replication-luatest/gh_5568_read_only_reason_test.lua

```diff
@@ -330,10 +329,13 @@ g.test_read_only_reason_synchro_no_uuid = function(g)
 
     local leader_id = g.master:instance_id()
     t.helpers.retrying({}, function()
-        assert(g.replica:exec(function(leader_id)
-            return box.info.synchro.queue.owner ~= 0 and
-                   box.info.replication[leader_id] == nil
-        end, {leader_id}))
+        t.assert_not_equals(g.replica:exec(function()
+            return box.info.synchro.queue.owner
+        end), 0)
+
+        t.assert_equals(g.replica:exec(function(leader_id)
+            return box.info.replication[leader_id]
+        end, {leader_id}), nil)
     end)
```

Comment:
I don't know for sure, but maybe those two conditions are checked without yeilds in between intentionally.